buildstep_inspect.py

```diff
@@ -0,0 +1,60 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from __future__ import unicode_literals
+from __future__ import print_function
+
+from atomic_reactor.plugin import BuildStepPlugin
+from atomic_reactor.build import BuildResult
+
+
+class BuildStepInspectPlugin(BuildStepPlugin):
+    """
+    Log the inputs to the build-step plugin
+    """
+
+    key = 'buildstep_inspect'
+    is_allowed_to_fail = True
```

Comment:
I've added arg 'inappropriate': set it to True and the plugin will raise InappropriateBuildStepError.